src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -0,0 +1,182 @@
+
+        /// <summary>
+        /// Meter to hold the MSAL metrics.
+        /// </summary>
+        internal static readonly Meter Meter = new Meter(MeterName, "1.0.0");
+
+        /// <summary>
+        /// ActivitySource to hold the MSAL activities.
+        /// </summary>
+        internal static readonly ActivitySource s_acquireTokenActivity = new ActivitySource(ActivitySourceName, "1.0.0");
+
+        /// <summary>
+        /// Counter to hold the number of successful token acquisition calls.
+        /// </summary>
+        internal static readonly Lazy<Counter<long>> s_successCounter = new(() => Meter.CreateCounter<long>(
+            SuccessCounterName,
+            description: "Number of successful token acquisition calls"));
+
+        /// <summary>
+        /// Counter to hold the number of failed token acquisition calls.
+        /// </summary>
+        internal static readonly Lazy<Counter<long>> s_failureCounter = new(() => Meter.CreateCounter<long>(
+            FailedCounterName,
+            description: "Number of failed token acquisition calls"));
+
+        /// <summary>
+        /// Histogram to record total duration of token acquisition calls.
+        /// </summary>
+        internal static readonly Histogram<long> s_durationTotal = Meter.CreateHistogram<long>(
+            TotalDurationHistogramName,
+            unit: "ms",
+            description: "Performance of token acquisition calls total latency");
+
+        /// <summary>
+        /// Histogram to record total duration of token acquisition calls in microseconds(us) when total duration in ms is less than 5.
+        /// This will capture the duration in us when the L1 cache is used to get the token.
+        /// </summary>
+        internal static readonly Histogram<long> s_durationTotalInUs = Meter.CreateHistogram<long>(
+            TotalDurationInUsHistogramName,
+            unit: "us",
+            description: "Performance of token acquisition calls total latency in micro seconds when L1 cache is used.");
+
+        /// <summary>
+        /// Histogram to record duration in cache of token acquisition calls.
+        /// </summary>
+        internal static readonly Histogram<long> s_durationInCache = Meter.CreateHistogram<long>(
+            DurationInCacheHistogramName,
+            unit: "ms",
+            description: "Performance of token acquisition calls cache latency");
+
+        /// <summary>
+        /// Histogram to record duration in http of token acquisition calls.
+        /// </summary>
+        internal static readonly Histogram<long> s_durationInHttp = Meter.CreateHistogram<long>(
+            DurationInHttpHistogramName,
+            unit: "ms",
+            description: "Performance of token acquisition calls network latency");
+
+        internal static readonly Lazy<Activity> s_activity = new Lazy<Activity>(() => s_acquireTokenActivity.StartActivity("Token Acquisition", ActivityKind.Internal));
```

Comment:
What is the intended lifetime of Otel `Activity`? Is it equivalent to a request, ie per unique correlation ID? If so , this shouldn't be static? Otherwise, there would be race conditions, since multiple parallel requests could set their own tags on here.